### DISCUSSION TITLE
How to Converting L2（metric type） Scores to Similarity scores

### DISCUSSION THREAD

**Discussion by littlePoBoy** [milvus-io/milvus#35485] [Q&A and General discussion]
When using L2 (Euclidean distance) as the metric type in Milvus，the returned scores are actually squared L2 distances，the distance range is [0.0, +∞]，How to Converting Similarity scores[0,1] ，1 is the most similar。can you give me example..
anyway,thanks.

**Comment by yhmo** [comment]
When we compare two human faces, do we have a standard to say they are 80% similar or 60% similar? No such standard. Maybe I say they are 85% similar, but another people will say they are 75% similar.

Even if we define a standard to convert distance to score, different embedding models will get different scores at a certain distance value. Just take a look this chart then you will understand what I mean:
![distance_vs_score drawio](https://github.com/user-attachments/assets/6f40192d-71ab-4b9f-af32-3a5880ab5cb9)

Read this thread for more info:
https://github.com/milvus-io/milvus/discussions/32625
